tests/units/anta_tests/routing/test_bgp.py

```diff
@@ -1239,4 +1239,216 @@
             ],
         },
     },
+    {
+        "name": "success",
+        "test": VerifyBGPExchangedRoutes,
+        "eos_data": [
+            {
+                "vrfs": {
+                    "default": {
+                        "bgpRouteEntries": {
+                            "192.0.254.3/32": {
+                                "bgpRoutePaths": [
+                                    {
+                                        "routeType": {
+                                            "valid": True,
+                                            "active": True,
+                                        },
+                                    }
+                                ]
+                            },
+                        },
+                    }
+                }
+            },
+            {
+                "vrfs": {
+                    "default": {
+                        "bgpRouteEntries": {
+                            "192.0.254.3/32": {
+                                "bgpRoutePaths": [
+                                    {
+                                        "routeType": {
+                                            "valid": True,
+                                            "active": True,
+                                        },
+                                    }
+                                ],
+                            }
+                        },
+                    }
+                }
+            },
+        ],
+        "inputs": {
+            "bgp_neighbors": [
+                {
+                    "neighbor": "172.30.11.1",
+                    "vrf": "default",
+                    "advertised_routes": ["192.0.254.3/32"],
+                    "received_routes": ["192.0.254.3/32"],
+                },
+            ]
+        },
+        "expected": {"result": "success"},
+    },
+    {
+        "name": "failure-no-routes",
+        "test": VerifyBGPExchangedRoutes,
+        "eos_data": [
+            {"vrfs": {"default": {"vrf": "default", "routerId": "192.0.255.1", "asn": "65001", "bgpRouteEntries": {}}}},
+            {"vrfs": {"default": {"vrf": "default", "routerId": "192.0.255.1", "asn": "65001", "bgpRouteEntries": {}}}},
```

Comment:
so both commands return the same structure everytime? Why do we run both?

Reply:
yes both command return same structure output but to check advertised or received routes for a neighbor we need to change the command. From single command we are not getting both advertised and received routes